### PR TITLE
AMLS-4887: Page split Is your registered office in the UK?

### PR DESCRIPTION
**Pull request by drewmiley**
## Checklist

- [ ] Breaking change (fix or feature that would cause existing functionality to change).
- [x] Build passing.
- [x] Unit tests have full coverage.
- [ ] Unit test approach and implementation has been reviewed with QA (testers).
- [x] Requires acceptance test run.
- [x] Appropriate labels added.
- [x] RELEASE NOTES ADDED.

### CODE REVIEW

**Comment by drewmiley** [comment]
Screenshot 2019-06-21 at 16 27 11

**Comment by drewmiley** [comment]
Also need to update views.businessdetails.summarySpec

File: app/controllers/businessdetails/RegisteredOfficeIsUKController.scala [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+          response =>
+            val form: Form2[RegisteredOfficeIsUK] = (for {
+              businessDetails <- response
+              registeredOffice <- businessDetails.registeredOffice
+            } yield registeredOffice match {
+              case _: RegisteredOfficeUK => Form2[RegisteredOfficeIsUK](RegisteredOfficeIsUK(true))
+              case _: RegisteredOfficeNonUK => Form2[RegisteredOfficeIsUK](RegisteredOfficeIsUK(false))
+            }) getOrElse EmptyForm
+            Ok(registered_office_is_uk(form, edit))
+
```
Comment (drewmiley) [inline app/controllers/businessdetails/RegisteredOfficeIsUKController.scala]: Remove line

File: app/controllers/businessdetails/RegisteredOfficeNonUKController.scala [hunk 2]
```diff
@@ -54,8 +52,8 @@ class RegisteredOfficeController @Inject () (
             val form: Form2[RegisteredOffice] = (for {
               businessDetails <- response
               registeredOffice <- businessDetails.registeredOffice
-            } yield Form2[RegisteredOffice](registeredOffice)).getOrElse(Form2[RegisteredOffice](preSelectUK))
-            Ok(registered_office(form, edit, autoCompleteService.getCountries))
+            } yield Form2[RegisteredOffice](registeredOffice)) getOrElse EmptyForm
+            Ok(registered_office_non_uk(form, edit, autoCompleteService.getCountries))
```
Comment (drewmiley) [inline app/controllers/businessdetails/RegisteredOfficeNonUKController.scala]: Remove line

File: app/controllers/businessdetails/RegisteredOfficeUKController.scala [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    implicit authContext =>
+      implicit request =>
+        dataCacheConnector.fetch[BusinessDetails](BusinessDetails.key) map {
+          response =>
+            val form: Form2[RegisteredOffice] = (for {
+              businessDetails <- response
+              registeredOffice <- businessDetails.registeredOffice
+            } yield Form2[RegisteredOffice](registeredOffice)) getOrElse EmptyForm
+            Ok(registered_office_uk(form, edit))
+
```
Comment (drewmiley) [inline app/controllers/businessdetails/RegisteredOfficeUKController.scala]: Remove line

File: app/controllers/businessdetails/RegisteredOfficeNonUKController.scala [hunk 4]
```diff
@@ -64,7 +62,7 @@ class RegisteredOfficeController @Inject () (
     implicit authContext => implicit request =>
         Form2[RegisteredOffice](request.body) match {
           case f: InvalidForm =>
-            Future.successful(BadRequest(registered_office(f, edit, autoCompleteService.getCountries)))
+            Future.successful(BadRequest(registered_office_non_uk(f, edit, autoCompleteService.getCountries)))
           case ValidForm(_, data) =>
 
             val doUpdate = for {
```
Comment (drewmiley) [inline app/controllers/businessdetails/RegisteredOfficeNonUKController.scala]: Look ar routing

File: app/controllers/businessdetails/RegisteredOfficeUKController.scala [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+            Future.successful(BadRequest(registered_office_uk(f, edit)))
+          case ValidForm(_, data) =>
+
+            val doUpdate = for {
+              businessDetails <- OptionT(dataCacheConnector.fetch[BusinessDetails](BusinessDetails.key))
+              _ <- OptionT.liftF(dataCacheConnector.save[BusinessDetails](BusinessDetails.key, businessDetails.registeredOffice(data)))
+              status <- OptionT.liftF(statusService.getStatus)
+              _ <- OptionT.liftF(auditAddressChange(data, businessDetails.registeredOffice, edit)) orElse OptionT.some(Success)
+            } yield {
+              if (redirectToDateOfChange[RegisteredOffice](status, businessDetails.registeredOffice, data)) {
```
Comment (drewmiley) [inline app/controllers/businessdetails/RegisteredOfficeUKController.scala]: Look at routing

File: test/controllers/businessdetails/RegisteredOfficeControllerNonUKSpec.scala [hunk 6]
```diff
@@ -0,0 +1,259 @@
+/*
```
Comment (drewmiley) [inline test/controllers/businessdetails/RegisteredOfficeControllerNonUKSpec.scala]: Rename

File: test/controllers/businessdetails/RegisteredOfficeControllerUKSpec.scala [hunk 7]
```diff
@@ -38,18 +38,17 @@ import utils.{AmlsSpec, AuthorisedFixture, AutoCompleteServiceMocks}
 import scala.collection.JavaConversions._
 import scala.concurrent.Future
 
-class RegisteredOfficeControllerSpec extends AmlsSpec with  MockitoSugar{
+class RegisteredOfficeControllerUKSpec extends AmlsSpec with  MockitoSugar{
```
Comment (drewmiley) [inline test/controllers/businessdetails/RegisteredOfficeControllerUKSpec.scala]: Rename

File: app/controllers/businessdetails/RegisteredOfficeUKController.scala [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+  }
+
+  def post(edit: Boolean = false) = Authorised.async {
+    implicit authContext => implicit request =>
+        Form2[RegisteredOffice](request.body) match {
+          case f: InvalidForm =>
+            Future.successful(BadRequest(registered_office_uk(f, edit)))
+          case ValidForm(_, data) =>
+
+            val doUpdate = for {
```
Comment (slavvco) [inline app/controllers/businessdetails/RegisteredOfficeUKController.scala]: not sure if doUpdate describes well what it does

File: app/controllers/businessdetails/RegisteredOfficeUKController.scala [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+              _ <- OptionT.liftF(dataCacheConnector.save[BusinessDetails](BusinessDetails.key, businessDetails.registeredOffice(data)))
+              status <- OptionT.liftF(statusService.getStatus)
+              _ <- OptionT.liftF(auditAddressChange(data, businessDetails.registeredOffice, edit)) orElse OptionT.some(Success)
+            } yield {
+              if (redirectToDateOfChange[RegisteredOffice](status, businessDetails.registeredOffice, data)) {
+                Redirect(routes.RegisteredOfficeDateOfChangeController.get())
+              } else {
+                edit match {
+                  case true => Redirect(routes.SummaryController.get())
+                  case _ => Redirect(routes.ContactingYouController.get(edit))
```
Comment (owen-opencast) [inline app/controllers/businessdetails/RegisteredOfficeUKController.scala]: Edit is always false here?

File: conf/businessDetails.routes [hunk 10]
```diff
@@ -16,8 +16,14 @@ POST        /corporation-tax                                       @controllers.
 GET         /registered-office-address                             @controllers.businessdetails.ConfirmRegisteredOfficeController.get(edit: Boolean ?= false)
 POST        /registered-office-address                             @controllers.businessdetails.ConfirmRegisteredOfficeController.post(edit: Boolean ?= false)
 
-GET         /business-address                                      @controllers.businessdetails.RegisteredOfficeController.get(edit: Boolean ?= false)
-POST        /business-address                                      @controllers.businessdetails.RegisteredOfficeController.post(edit: Boolean ?= false)
+GET         /business-address                                      @controllers.businessdetails.RegisteredOfficeIsUKController.get(edit: Boolean ?= false)
```
Comment (owen-opencast) [inline conf/businessDetails.routes]: Should be registered-office-country?

File: app/controllers/businessdetails/RegisteredOfficeUKController.scala (hunk 9 above)
Comment (drewmiley) [reply app/controllers/businessdetails/RegisteredOfficeUKController.scala]: Done for both controllers

File: conf/businessDetails.routes (hunk 10 above)
Comment (drewmiley) [reply conf/businessDetails.routes]: Good spot.

File: app/controllers/businessdetails/RegisteredOfficeUKController.scala (hunk 8 above)
Comment (drewmiley) [reply app/controllers/businessdetails/RegisteredOfficeUKController.scala]: Removed this

File: conf/businessDetails.routes (hunk 10 above)
Comment (drewmiley) [reply conf/businessDetails.routes]: Done

**Pull request merged**